packages/sit-onyx/src/components/OnyxDataGrid/types.ts

```diff
@@ -101,6 +103,18 @@ export type OnyxDataGridProps<
  * "Raw" user data for a data grid entry/row, e.g. fetched from a backend service.
  */
 export type DataGridEntry = {
-  id: PropertyKey;
   [key: PropertyKey]: unknown;
+  /**
+   * Unique ID of the data grid entry/row.
+   */
+  id: PropertyKey;
+  /**
+   * Attributes that are bound directly to the `<tr>` element of the row.
+   */
+  _trAttributes?: HTMLAttributes;
+  /**
+   * Overrides which columns to render in which order.
+   * Useful if e.g. adding a custom full-width row inside a data grid feature.
+   */
+  _columns?: Omit<InternalColumnConfig<{ id: PropertyKey }>, "label">[];
```

Comment:
For these kind of APIs, where "magic properties" are expected, Symbols should be used:

```suggestion
  [RowOverrideSymbol]: {
    _trAttributes?: HTMLAttributes;
    /**
     * Overrides which columns to render in which order.
     * Useful if e.g. adding a custom full-width row inside a data grid feature.
     */
    _columns?: Omit<InternalColumnConfig<{ id: PropertyKey }>, "label">[];
  }
```

But IMHO the data input should not have these side-effects, instead it should be done matching the row ID and enhancing/changing the data in a feature.